File: Programmers/programmers_로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int cant_recog = 0;
    int same = 0;
    for(auto num:lottos){
        if(num==0){
            cant_recog++; continue;
        }
        for(auto winNum:win_nums){
            if(winNum==num){
                same++; break;
            }
        }
    }
    int best = 7-(cant_recog+same);
    int worst = 7-(same);
    if(best>=6) answer.push_back(6);
    else answer.push_back(best);
    if(worst>=6) answer.push_back(6);
    else answer.push_back(worst);
    return answer;
}

File: Programmers/programmers_로또의최고순위와최저순위.py
def solution(lottos, win_nums):
    #1:6개 일치, 2:5개 일치, ,,, 5:2개 일치, 6:1개혹은 0개일치
    rank = [6, 6, 5, 4, 3, 2, 1] #rank[0]는 0개 일치 했을때 6등이라는 의미.
    same = 0
    zeros = lottos.count(0)
    for num in win_nums:
        if num in lottos :
            same += 1
    answer = []
    answer.append(rank[zeros+same])
    answer.append(rank[same])
    
    return answer
